Sources/PortalSwift/Core/Api/PortalSwapLifiApi.swift

```diff
@@ -0,0 +1,144 @@
+  ///   - requests: An instance of PortalRequestsProtocol to handle HTTP requests.
+  public init(
+    apiKey: String,
+    apiHost: String = "api.portalhq.io",
+    requests: PortalRequestsProtocol? = nil
+  ) {
+    self.apiKey = apiKey
+    self.baseUrl = apiHost.starts(with: "localhost") ? "http://\(apiHost)" : "https://\(apiHost)"
+    self.requests = requests ?? PortalRequests()
+  }
+
+  /*******************************************
+   * Public functions
+   *******************************************/
+
+  /// Retrieves routes from the Lifi integration.
+  public func getRoutes(request: LifiRoutesRequest) async throws -> LifiRoutesResponse {
+    guard let url = URL(string: "\(baseUrl)/api/v3/clients/me/integrations/lifi/routes") else {
+      logger.error("PortalSwapLifiApi.getRoutes() - Unable to build request URL.")
+      throw URLError(.badURL)
+    }
+
+    do {
+      return try await post(url, withBearerToken: apiKey, andPayload: request, mappingInResponse: LifiRoutesResponse.self)
+    } catch {
+      logger.error("PortalSwapLifiApi.getRoutes() - Error: \(error.localizedDescription)")
+      throw error
+    }
+  }
+
+  /// Retrieves a quote from the Lifi integration.
+  public func getQuote(request: LifiQuoteRequest) async throws -> LifiQuoteResponse {
+    guard let url = URL(string: "\(baseUrl)/api/v3/clients/me/integrations/lifi/quote") else {
+      logger.error("PortalSwapLifiApi.getQuote() - Unable to build request URL.")
+      throw URLError(.badURL)
+    }
+
+    do {
+      return try await post(url, withBearerToken: apiKey, andPayload: request, mappingInResponse: LifiQuoteResponse.self)
+    } catch {
+      logger.error("PortalSwapLifiApi.getQuote() - Error: \(error.localizedDescription)")
+      throw error
+    }
+  }
+
+  /// Retrieves the status of a transaction from the Lifi integration.
+  public func getStatus(request: LifiStatusRequest) async throws -> LifiStatusResponse {
+    var queryParams: [String] = []
+    addParam("fromChain", request.fromChain, to: &queryParams)
+    addParam("txHash", request.txHash, to: &queryParams)
+    addParam("bridge", request.bridge, to: &queryParams)
+    addParam("toChain", request.toChain, to: &queryParams)
+
+    let queryString = queryParams.isEmpty ? "" : "?\(queryParams.joined(separator: "&"))"
+
+    guard let url = URL(string: "\(baseUrl)/api/v3/clients/me/integrations/lifi/status\(queryString)") else {
+      logger.error("PortalSwapLifiApi.getStatus() - Unable to build request URL.")
+      throw URLError(.badURL)
+    }
+
+    do {
+      return try await get(url, withBearerToken: apiKey, mappingInResponse: LifiStatusResponse.self)
+    } catch {
+      logger.error("PortalSwapLifiApi.getStatus() - Error: \(error.localizedDescription)")
+      throw error
+    }
+  }
+
+  /// Submits step transaction details to the Lifi integration.
+  public func stepTransaction(request: LifiStepTransactionRequest) async throws -> LifiStepTransactionResponse {
```

Comment:
We should probably rename this method to something like `getRouteStep` or `getRouteStepDetails`